service/frontend/task_reachability.go

```diff
@@ -48,38 +53,36 @@ import (
 type versionSetFetcher struct {
 	lock           sync.Mutex
 	matchingClient matchingservice.MatchingServiceClient
-	futures        map[string]future.Future[*matchingservice.GetWorkerBuildIdCompatibilityResponse]
+	futures        map[string]future.Future[*persistencespb.VersioningData]
 }
 
 func newVersionSetFetcher(matchingClient matchingservice.MatchingServiceClient) *versionSetFetcher {
 	return &versionSetFetcher{
 		matchingClient: matchingClient,
-		futures:        make(map[string]future.Future[*matchingservice.GetWorkerBuildIdCompatibilityResponse]),
+		futures:        make(map[string]future.Future[*persistencespb.VersioningData]),
 	}
 }
 
-func (f *versionSetFetcher) getFuture(ctx context.Context, ns *namespace.Namespace, taskQueue string) future.Future[*matchingservice.GetWorkerBuildIdCompatibilityResponse] {
+func (f *versionSetFetcher) getFuture(ctx context.Context, ns *namespace.Namespace, taskQueue string) future.Future[*persistencespb.VersioningData] {
 	f.lock.Lock()
 	defer f.lock.Unlock()
 	_, found := f.futures[taskQueue]
 	if !found {
-		fut := future.NewFuture[*matchingservice.GetWorkerBuildIdCompatibilityResponse]()
+		fut := future.NewFuture[*persistencespb.VersioningData]()
 		f.futures[taskQueue] = fut
 		go func() {
-			value, err := f.matchingClient.GetWorkerBuildIdCompatibility(ctx, &matchingservice.GetWorkerBuildIdCompatibilityRequest{
-				NamespaceId: ns.ID().String(),
-				Request: &workflowservice.GetWorkerBuildIdCompatibilityRequest{
-					Namespace: ns.Name().String(),
-					TaskQueue: taskQueue,
-				},
+			value, err := f.matchingClient.GetTaskQueueUserData(ctx, &matchingservice.GetTaskQueueUserDataRequest{
+				NamespaceId:   ns.ID().String(),
+				TaskQueue:     taskQueue,
```

Comment:
this always goes to the root (it did before too). probably okay but we could consider load-balancing this

Reply:
I'm more concerned with hammering visibility than getting this info from matching but I do see your point.
Do you think it's worth it?

Reply:
nah, you're right this should be cheap compared to visibility